crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -1775,6 +1775,12 @@ impl<'a> StateWorkingSet<'a> {
     }
 
     pub fn find_variable(&self, name: &[u8]) -> Option<VarId> {
+        let name = if name.starts_with(b"$") {
+            &name[1..]
```

Comment:
Do we need to do this step?

If we always store variables without the dollar sign, then I think we can skip it. Though maybe you found an example that relies on it?

Reply:
I think we should fix them in the completer. That way the engine all works the same way, so when the user is waiting for the parser to finish, they aren't waiting for us to do a lot of repeating work that we can do one time.